src/Curves/CurveOperations.jl

```diff
@@ -0,0 +1,226 @@
+
+    point[1] = evalWithDict(ex, Dict(xArg=> t))
+    point[2] = evalWithDict(ey, Dict(yArg=> t))
+
+end
+
+
+function splineCurvePoints(nKnots::Int, splineData::Array{Float64,2}, points::Array{Float64,2})
+
+    xSpline = constructSpline(nKnots,splineData[:,1],splineData[:,2])
+    ySpline = constructSpline(nKnots,splineData[:,1],splineData[:,3])
+
+    sz   = size(points)
+    nPts = sz[1]
+    t = 0.0
+    for i = 1:nPts
+        t = (i-1)/(nPts-1)
+        points[i,1] = evalSpline(xSpline,t)
+        points[i,2] = evalSpline(ySpline,t)
+    end
+end
+
+
+function splineCurvePoint(nKnots::Int, splineData::Array{Float64,2}, t, point::Array{Float64})
+
+    xSpline = constructSpline(nKnots,splineData[:,1],splineData[:,2])
+    ySpline = constructSpline(nKnots,splineData[:,1],splineData[:,3])
+
+    point[1] = evalSpline(xSpline,t)
+    point[2] = evalSpline(ySpline,t)
+end
+
+
+# This function evaluates a string as an equation, might be redundant code
+# function parse_eval_dict(s::AbstractString, locals::Dict{Symbol})
+#     ex = Meta.parse(s)
+#     assignments = [:($sym = $val) for (sym,val) in locals]
+#     eval(:(let $(assignments...); $ex; end))
+# end
+
+
+function evalWithDict(ex, locals::Dict{Symbol})
+    assignments = [:($sym = $val) for (sym,val) in locals]
+    eval(:(let $(assignments...); $ex; end))
+end
+
+
+function curvePoints(crvDict::Dict{String,Any}, N::Int)
+# N = Number of intervals
+    curveType::String = crvDict["TYPE"]
+
+    if curveType == "PARAMETRIC_EQUATION_CURVE"
+        xEqn = crvDict["xEqn"]
+        yEqn = crvDict["yEqn"]
+
+        x = zeros(Float64,N+1,2)
+        t = zeros(Float64,N+1)
+        for i = 1:N+1
+            t[i] = (i-1)/N
+        end
+        peEquationCurvePoints(xEqn,yEqn,t,x)
+    elseif curveType == "END_POINTS_LINE"
+        xStart = realArrayForKeyFromDictionary("xStart",crvDict)
+        xEnd   = realArrayForKeyFromDictionary("xEnd",crvDict)
+        x = zeros(Float64,3,2)
+        t = zeros(Float64,3)
+        for i = 1:3
+            t[i] = (i-1)/2.0
+        end
+
+        endPointsLineCurvePoints(xStart,xEnd,t,x)
+    elseif curveType == "CIRCULAR_ARC"
+        center     = realArrayForKeyFromDictionary("center",crvDict)
+        radius     = realForKeyFromDictionary("radius",crvDict)
+        startAngle = realForKeyFromDictionary("start angle",crvDict)
+        endAngle   = realForKeyFromDictionary("end angle",crvDict)
+        units      = crvDict["units"]
+
+        x = zeros(Float64,N+1,2)
+        t = zeros(Float64,N+1)
+        for i = 1:N+1
+            t[i] = (i-1)/N
+        end
+
+        arcCurvePoints(center,radius,startAngle,endAngle,units,t,x)
+    elseif curveType == "SPLINE_CURVE"
+        nKnots = intForKeyFromDictionary("nKnots",crvDict)
+        splineData = crvDict["SPLINE_DATA"]
+
+        M = max(N,nKnots*2)
+        x = zeros(Float64,M+1,2)
+        t = zeros(Float64,M+1)
+        for i = 1:M+1
+            t[i] = (i-1)/M
+        end
```

Comment:
```suggestion
        x = zeros(Float64,M+1,2)
```
As far as I can tell, neither `M` or `t` are used after declaration.

Reply:
`M` is used to declare the `x` array but `t`is unused. I can clean this up

Reply:
Right, sorry. But `t` can be removed imho.